### ISSUE TITLE
Dialog doesn't have focus trapped

### ISSUE THREAD

**Issue by TGlide** [aviflombaum/shadcn-rails#21]
# Description
According to [WAI-ARIA](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/), focus should be trapped withing an open Dialog's content, meaning pressing Tab should not move the focus outside of the page. That is not the case currently.

## Steps to reproduce

1. Open a Dialog
2. Press Tab a couple times

**Comment by aviflombaum** [comment]
On it

**Comment by petrenkorf** [comment]
@aviflombaum any updates on this?
